pkg/networkservice/common/clientconn/metadata.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package clientconn allows storing grpc.ClientConnInterface stored in per Connection.Id metadata
+package clientconn
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+)
+
+type key struct{}
+
+// Store sets the grpc.ClientConnInterface stored in per Connection.Id metadata.
+func Store(ctx context.Context, isClient bool, cc grpc.ClientConnInterface) {
```

Comment:
It looks like we want to perform dial only from the client chain, so is there any need for the `isClient` flag here?

Reply:
Fixed